src/java.base/unix/native/libjava/java_props_md.c

```diff
@@ -488,8 +488,14 @@ GetJavaProperties(JNIEnv *env)
 #else
         sprops.user_home = pwent ? strdup(pwent->pw_dir) : NULL;
 #endif
-        if (sprops.user_home == NULL) {
-            sprops.user_home = "?";
+        if (sprops.user_home == NULL || sprops.user_home[0] == '\0' ||
+            sprops.user_home[1] == '\0') {
+            // If the OS supplied home directory is not defined or less than two characters long
+            // $HOME is the backup source for the home directory, if defined
+            char* user_home = getenv("HOME");
+            if ((user_home != NULL) && (user_home[0] != '\0')) {
+                sprops.user_home = user_home;
+            }
```

Comment:
Is there any possibility where `user.home` is not initialized, and later causes SEGV or NPE? I just wonder the previous version always init to `?` which is odd, but guaranteed not to cause those errors.

Reply:
I can't imagine it not being set. But it is easier to track down the source of a "?" than the source of null.
I thought of changing from "?" to "UNKNOWN" or "NOHOMEDIR" or something but it seems quite remote
after adding the fallback to $HOME.

Reply:
Another option is an empty string `""`, but I don't have a strong preference if it won't throw any errors.